nixos/modules/services/system/kerberos/heimdal.nix

```
@@ -66,7 +66,7 @@ in
         };
       });
 
-    systemd.services.kadmind = {
+    systemd.services.kadmind = lib.mkIf cfg.roles.administrationServer {
```

```suggestion
    systemd.services.kadmind = {
      enable = cfg.roles.administrationServer;
```

Genearlly I prefer these since it means the unit is always there in the configuration data in case you want to reference it. It doesn't make a difference in the built system.

Though, both varaints are used in nixpkgs so it doesn't really matter.

---

Actually I really like your way!